text/0000-tcp-backpressure.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: TCP backpressure
+- Start Date: 2016-09-29
+- RFC PR: 
+- Pony Issue: 
+
+# Summary
+
+Add support for TCP backpressure to Pony's `TCPConnection` actor.
+
+# Motivation
+
+TCP provides backpressure to signal to applications that they need to slow down sending, however, currently Pony while itself being signaled that backpressure is being applied, doesn't surface that to application code. This means that even a well meaning, diligent programmer is unable to respond to backpressure by slowing down.
+
+# Detailed design
+
+Add 1 or 2 notification methods to `TCPConnectionNotify`. In the version we are running at Sendence, we added a single notifier method:
+
+```pony
+  fun ref throttled(conn: TCPConnection ref, status: Bool) =>
+    """
+    Called when there is a change in backpressure status of the connection
+    """
+    None
+```
+
+The method, like all other `TCPConnectionNotify` methods take a reference to the `TCPConnection` that has had a chance in backpressure status as well as a bool indicating if the connection is being throttled.
+
+While we are using this single method at Sendence, I think there is a strong argument for using 2 methods: one for when backpressure is being applied and one for when its released. While these shouldn't be called all that often, in `TCPConnection` we know whether this is a backpressure release or application but we are requiring the programming to do an branch to determine what to do in the `throttled` method. That's extra instructions that we don't really need. Additionally, the boolean for status is a little nasty and not the greatest API.
+
+Within `TCPConnection` we have added the following private methods:
+
+```pony
+  fun ref _apply_backpressure() =>
+    ifdef not windows then
+      _writeable = false
+    end
+
+    _notify.throttled(this, true)
+
+  fun ref _release_backpressure() =>
+    _notify.throttled(this, false)
+```
+
+For non Windows platforms:
+
+`_apply_backpressure` is called `write_final`
```

Comment:
This confused me a bit - based on the context from the next few examples I think you mean "is called in `write_final`" rather than "is called `write_final`"